lib/puppet/pops/types/type_formatter.rb

```diff
@@ -206,14 +206,26 @@ def string_PRuntimeType(t)
     append_array('Runtime', [string(t.runtime), string(t.runtime_type_name)])
   end
 
+  def is_empty_range?(from, to)
+    from == 0 && to == 0
+  end
+
   # @api private
   def string_PArrayType(t)
-    append_array('Array', t == PArrayType::DATA ? EMPTY_ARRAY : [string(t.element_type)] + range_array_part(t.size_type))
+    if t.has_empty_range?
+      'Array[0, 0]'
```

Comment:
Would prefer append_array('Array', ['0', '0']) to keep all array formatting in one place.